src/NHibernate/Hql/Ast/ANTLR/Tree/EntityJoinJoinSequenceImpl.cs

```diff
@@ -30,12 +30,14 @@ public EntityJoinJoinSequenceImpl(ISessionFactoryImplementor factory, EntityType
 		{
 			var joinFragment = new ANSIJoinFragment();
 
-			var on = withClauseFragment ?? new SqlString();
-			var filters = _entityType.GetOnCondition(_tableAlias, Factory, enabledFilters);
-			if (!string.IsNullOrEmpty(filters))
-			{
-				on.Append(" and ").Append(filters);
-			}
+			var on = withClauseFragment ?? SqlString.Empty;
+			//Note: filters logic commented as GetOnCondition always returns empty string for entity join (as IsReferenceToPrimaryKey is always true)
+			//TODO: If filters for entity join really make sense we need to inline GetOnCondition part that retrieves filters
+//			var filters = _entityType.GetOnCondition(_tableAlias, Factory, enabledFilters);
+//			if (!string.IsNullOrEmpty(filters))
+//			{
+//				on = on.Append(filters);
+//			}
```

Comment:
Unrelated change, should be reverted.

Reply:
Original code is non functional and `SqlString.Append` related:
```C#
if (!string.IsNullOrEmpty(filters))
{
	on.Append(" and ").Append(filters);
}
```

as `SqlString` is immutable all those appends make no sense. So I figure better to comment non functional code with some details how to fix it in the future.

Reply:
I agree that this change should be reverted to keep changes purely focused on SqlString usage changes.

Reply:
As I already said it's exactly only about `SqlString.Append`. Though yeah added comment does look like something unrelated - I will adjust comment to make it clear.